Sources/Datadog/Core/Persistence/Files/File.swift

```diff
@@ -43,23 +43,27 @@ internal struct File: WritableFile, ReadableFile {
     }
 
     /// Appends given data at the end of this file.
-    func append(transaction: ((Data) throws -> Void) throws -> Void) throws {
+    func append(data: Data) throws {
         let fileHandle = try FileHandle(forWritingTo: url)
-        defer { fileHandle.closeFile() }
 
-        try objcExceptionHandler.rethrowToSwift {
-            fileHandle.seekToEndOfFile()
-        }
+        if #available(iOS 13.0, *) {
+            try fileHandle.seekToEnd()
+            try fileHandle.write(contentsOf: data)
+            try fileHandle.close()
+        } else {
+            defer {
+                try? objcExceptionHandler.rethrowToSwift {
+                    fileHandle.closeFile()
+                }
+            }
 
-        // Writes given data at the end of the file.
-        func appendData(_ data: Data) throws {
             try objcExceptionHandler.rethrowToSwift {
-                fileHandle.write(data)
+                fileHandle.seekToEndOfFile()
             }
-        }
 
-        try transaction { chunkOfData in
-            try appendData(chunkOfData)
+            try objcExceptionHandler.rethrowToSwift {
+                fileHandle.write(data)
+            }
```

Comment:
we can put `fileHandle.seekToEndOfFile()` and `write(data)` into the same `rethrowToSwift` block, wdyt?

Reply:
👍